src/proxy.c

```diff
@@ -769,14 +847,30 @@ int main(int argc, char *argv[]) {
     }
 
 #ifdef LOG_TO_FILE
-    /* Set log handler for e.g. g_message(), g_warning() etc. */
+    /* Set log handler for e.g. g_message(), g_warning() etc, so we log to file */
     if (!open_log_file()) {
         g_error("Could not open log file\n");
         exit(1);
     }
-    g_log_set_handler(NULL, G_LOG_LEVEL_MASK, log_handler, NULL);
+    g_log_set_handler(NULL /*use default log domain*/,
+                      G_LOG_LEVEL_MASK,
+                      log_handler,
+                      NULL /*no need to pass data to handler*/);
 #endif
 
+    /* Default behavior is to be silent, to avoid issues with stdout not
+       being handled */
+    gboolean silence_logging = TRUE;
+#ifdef LOG_TO_STDOUT
+    silence_logging = FALSE;
+#endif
+    if (silence_logging) {
+        g_log_set_handler(NULL /*use default log domain*/,
+                          G_LOG_LEVEL_MASK,
+                          log_handler_silent,
+                          NULL /*no need to pass data to handler*/);
+    }
+
```

Comment:
maybe getting rid of a new variable would be nicer. So instead `#ifdef LOG_TO_STDOUT` you can add 
```
#ifndef LOG_TO_STDOUT
    g_log_set_handler(NULL, G_LOG_LEVEL_MASK, log_handler_silent, NULL);
#endif
```